internal/runtime/client/client.go

```diff
@@ -192,22 +194,57 @@ func (c *client) CallAllExtensions(ctx context.Context, hook runtimecatalog.Hook
 			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
 		}
 		tmpResponse := responseObject.(runtimehooksv1.ResponseObject)
+
 		err = c.CallExtension(ctx, hook, registration.Name, request, tmpResponse)
 		// If one of the extension handlers fails lets short-circuit here and return early.
 		if err != nil {
 			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
 		}
 		responses = append(responses, tmpResponse)
 	}
-	if err := aggregateResponses(responses); err != nil {
+
+	// Aggregate all responses into a single response.
+	aggregateSuccessfulResponses(response, responses)
+
+	if err != nil {
 		return errors.Wrap(err, "failed to aggregate responses")
 	}
 	return nil
 }
 
-func aggregateResponses(responses []runtimehooksv1.ResponseObject) error {
-	// TODO:(killianmuldoon) implement proper aggregation logic.
-	return nil
+func aggregateSuccessfulResponses(aggregatedResponse runtimehooksv1.ResponseObject, responses []runtimehooksv1.ResponseObject) {
+	// At this point the Status should always be ResponseStatusSuccess and the Message should be empty.
+	// So let's set those values to avoid keeping values that could have been set by the caller of CallAllExtensions.
+	aggregatedResponse.SetMessage("")
+	aggregatedResponse.SetStatus(runtimehooksv1.ResponseStatusSuccess)
+
+	if _, ok := aggregatedResponse.(runtimehooksv1.RetryResponseObject); !ok {
+		// If the aggregated response is not a RetryResponseObject then we're done.
+		return
+	}
+
+	aggregatedRetryableResponse := aggregatedResponse.(runtimehooksv1.RetryResponseObject)
+
+	for _, resp := range responses {
+		aggregatedRetryableResponse.SetRetryAfterSeconds(lowestNonZeroRetryAfterSeconds(
+			aggregatedRetryableResponse.GetRetryAfterSeconds(),
+			resp.(runtimehooksv1.RetryResponseObject).GetRetryAfterSeconds(),
```

Comment:
Can you add a comment somewhere here that it is okay for us to assume that the `resp` object will be retryable as long as the `aggregatedResponse` is retryable because they always will be of the same type?

Reply:
Done